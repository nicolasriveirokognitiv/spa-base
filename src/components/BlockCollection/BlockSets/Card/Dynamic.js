import React from 'react';

function Card() {
    return <div data-testid="image-with-text-dynamic">Dynamic Card</div>;
}

export default Card;
