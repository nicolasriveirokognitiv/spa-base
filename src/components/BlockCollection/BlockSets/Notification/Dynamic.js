import React from 'react';

function Notification() {
    return <div data-testid="notification-dynamic">Dynamic Notification</div>;
}

export default Notification;
