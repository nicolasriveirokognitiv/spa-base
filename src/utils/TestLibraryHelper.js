/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
