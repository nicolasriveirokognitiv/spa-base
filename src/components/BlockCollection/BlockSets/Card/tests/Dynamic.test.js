import React from 'react';
import { render } from 'Utils/TestLibraryHelper';
import Dynamic from '../Dynamic';

describe('Card Dynamic', () => {
    test('Should render Correctly', () => {
        expect(<Dynamic />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Dynamic />);
        expect(asFragment()).toMatchSnapshot();
    });
});
