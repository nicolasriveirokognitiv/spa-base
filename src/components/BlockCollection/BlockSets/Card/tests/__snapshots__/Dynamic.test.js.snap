// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Dynamic Should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="image-with-text-dynamic"
  >
    Dynamic Card
  </div>
</DocumentFragment>
`;
